internal/provider/cached_image_resource.go

```diff
@@ -293,6 +293,124 @@ func (r *CachedImageResource) Configure(ctx context.Context, req resource.Config
 	r.client = client
 }
 
+// setComputedEnv sets data.Env and data.EnvMap based on the values of the
+// other fields in the model.
+func (data *CachedImageResourceModel) setComputedEnv(ctx context.Context) diag.Diagnostics {
+	env := make(map[string]string)
+
+	env["ENVBUILDER_CACHE_REPO"] = tfValueToString(data.CacheRepo)
```

Comment:
Interesting edge-case, what if extra env overrides cache repo? We use `data.CacheRepo` directly elsewhere, but it may be different in computed env.

We should probably always prefer the values from computed env when running the cache probe?

Reply:
> We should probably always prefer the values from computed env when running the cache probe?

I don't agree. The provider should use the arguments it is supplied. We add a warning diag if you override existing environment variables set, which should be enough of an indication.

Alternatively, I could add a check for this and make it a diag.Error?

Reply:
I'm hesitant to block it with an error, so I'm just going to not allow overriding these but warn if they are set.